### PR TITLE
New Frontpage & feed

### PR DESCRIPTION
**Pull request by andygray**
@andygray to test - DO NOT MERGE

### CODE REVIEW

File: src/components/ui-controls/badges/IpfsBadge.vue [hunk 1]
```diff
@@ -1,13 +1,20 @@
 <template>
      <span class="badge badge-pill badge-extra-data" v-if="edition.tokenUri">
-      <img src="../../../../static/ipfs-logo.png" style="height: 11px"/> <a :href="edition.tokenUri" target="_blank">IPFS</a>
+       <a :href="edition.tokenUri" target="_blank">
```
Comment (jamesmorgan) [inline src/components/ui-controls/badges/IpfsBadge.vue]: Dead code `:href="edition.tokenUri"`

File: src/components/ui-controls/cards/GalleryCard.vue [hunk 2]
```diff
@@ -34,7 +34,7 @@
           </p>
         </div>
       </div>
-      <router-link :to="{ name: 'artist-v2', params: { artistAccount: edition.artistAccount } }" class="floating-artist-link">
+      <router-link :to="{ name: 'artist-v2', params: { artistAccount: edition.artistAccount } }" class="floating-artist-link" v-if="findArtistsForAddress(edition.artistAccount)">
```
Comment (jamesmorgan) [inline src/components/ui-controls/cards/GalleryCard.vue]: `v-if="findArtistsForAddress(edition.artistAccount)"` not sure why this is needed, all artists should be know to the app?

Comment (andygray) [reply src/components/ui-controls/cards/GalleryCard.vue]: fails on start - once loaded fine

File: src/components/pages/Feed.vue [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+      Availability,
+      ClickableAddress,
+      ArtistPanel,
+      HighResLabel,
+      SimpleArtistPanel
+    },
+    data () {
+      return {
+        PAGES,
+        LATEST: [
```
[the remark this answers is not in the document]
Comment (jamesmorgan) [reply src/components/pages/Feed.vue]: we can run a query for them

File: src/components/pages/Gallery.vue [hunk 4]
```diff
@@ -83,31 +58,34 @@
...
+      loadMoreEditions: function () {
+        this.$store.dispatch(`kodaV2/${actions.LOAD_MORE_EDITIONS}`);
       }
     },
     computed: {
+      ...mapState('kodaV2', [
+        'assets',
+        'assetsByTypeShuffled',
+        'assetsByTypeStart',
+        'assetsByTypeEnd'
```
Comment (jamesmorgan) [inline src/components/pages/Gallery.vue]: are these all used still?

File: src/components/pages/Gallery.vue [hunk 5]
```diff
@@ -83,31 +58,34 @@
...
+      ]),
       ...mapGetters('kodaV2', [
         'filterEditions',
-        'featuredArtistAccount'
-      ]),
-      ...mapGetters([
-        'findArtistsForAddress'
+        'editionsArray'
```
[the remark this answers is not in the document]
Comment (jamesmorgan) [reply src/components/pages/Gallery.vue]: maybe rename as not just an array

File: src/components/ui-controls/cards/GalleryCard.vue (hunk 2 above)
Comment (jamesmorgan) [reply src/components/ui-controls/cards/GalleryCard.vue]: maybe we can ditch this now

File: src/components/ui-controls/generic/LoadingSection.vue [hunk 6]
```diff
@@ -1,15 +1,12 @@
...
-
-    <div class="row justify-content-sm-center" v-if="isLoading(page)">
-      <div class="col text-center mt-5">
-        <p>We are loading assets from the Blockchain.</p>
-        <p>Please be patient as we are fully decentralised.</p>
-      </div>
-    </div>
+  <div class="text-center">
+    <loading-spinner v-if="isLoading(page)" class="mt-5"></loading-spinner>
+    <!--<div class="row justify-content-sm-center" v-if="isLoading(page)">-->
```
[the remark this answers is not in the document]
Comment (jamesmorgan) [reply src/components/ui-controls/generic/LoadingSection.vue]: yep

File: src/store/actions.js [hunk 7]
```diff
@@ -17,6 +17,7 @@ export const LOAD_EDITIONS_FOR_TYPE = 'LOAD_EDITIONS_FOR_TYPE';
 export const LOAD_EDITIONS_FOR_ARTIST = 'LOAD_EDITIONS_FOR_ARTIST';
 export const LOAD_INDIVIDUAL_EDITION = 'LOAD_INDIVIDUAL_EDITION';
 export const LOAD_INDIVIDUAL_TOKEN = 'LOAD_INDIVIDUAL_TOKEN';
+export const LOAD_MORE_EDITIONS = 'LOAD_MORE_EDITIONS';
```
Comment (jamesmorgan) [inline src/store/actions.js]: dead?

File: src/store/modules/kodaV2.js [hunk 8]
```diff
@@ -105,12 +95,23 @@ const contractStateModule = {
...
+
+      return artistEditions;
+    },
+    editionsArrayForArtist: (state, getters) => (artistAccount) => {
+      let editionsObjForArtist = getters.editionsForArtist(artistAccount);
+      return _.orderBy(_.values(editionsObjForArtist), 'edition', ['desc']);
+    },
+    editionsArray: (state) => () => {
+      if (!state.assets) {
+        return false;
```
Comment (jamesmorgan) [inline src/store/modules/kodaV2.js]: false?

File: src/store/modules/kodaV2.js [hunk 9]
```diff
@@ -123,32 +124,35 @@ const contractStateModule = {
...
         state.accountOwnedEditions.splice(index, 0, edition);
       }
     },
-    [mutations.SET_EDITION](state, data) {
+    [mutations.SET_EDITION] (state, data) {
       setEditionData(data, state);
     },
-    [mutations.SET_EDITIONS](state, editions) {
+    [mutations.SET_EDITIONS] (state, editions) {
+      console.time('SET_EDITIONS');
```
Comment (jamesmorgan) [inline src/store/modules/kodaV2.js]: remove

File: src/store/modules/kodaV2.js [hunk 10]
```diff
@@ -123,32 +124,35 @@ const contractStateModule = {
...
       _.forEach(editions, (data) => {
+        // console.log(data);
         setEditionData(data, state);
       });
+      console.timeEnd('SET_EDITIONS');
```
Comment (jamesmorgan) [inline src/store/modules/kodaV2.js]: remove

File: src/components/pages/Gallery.vue (hunk 4 above)
Comment (andygray) [reply src/components/pages/Gallery.vue]: removed

File: src/components/ui-controls/generic/LoadingSection.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (andygray) [reply src/components/ui-controls/generic/LoadingSection.vue]: 👍

File: src/store/actions.js (hunk 7 above)
Comment (andygray) [reply src/store/actions.js]: removed

File: src/store/modules/kodaV2.js (hunk 8 above)
Comment (andygray) [reply src/store/modules/kodaV2.js]: just return?

File: src/store/modules/kodaV2.js (hunk 9 above)
Comment (andygray) [reply src/store/modules/kodaV2.js]: I'd leave so we can monitor

File: src/components/pages/ArtistPage.vue [hunk 11]
```diff
@@ -126,12 +130,12 @@
...
       },
     },
-    created() {
+    created () {
       this.$store.dispatch(`loading/${actions.LOADING_STARTED}`, PAGES.ARTISTS);
 
       const loadData = function () {
-        this.$store.dispatch(`kodaV2/${actions.LOAD_EDITIONS_FOR_ARTIST}`, {artistAccount: this.getArtistAddress()})
-          .finally(() => {
+        this.$store.dispatch(`kodaV2/${actions.LOAD_EDITIONS_FOR_TYPE}`, {editionType: 1})
```
[the remark this answers is not in the document]
Comment (jamesmorgan) [reply src/components/pages/ArtistPage.vue]: Not sure this is needed - why not only load for artist

File: src/components/pages/Feed.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (jamesmorgan) [reply src/components/pages/Feed.vue]: I have done this

File: src/store/modules/kodaV2.js [hunk 12]
```diff
@@ -319,11 +315,16 @@ const loadEditionData = async (contract, edition) => {
...
+      highResAvailable: isHighRes(ipfsData, editionNumber)
     };
   } catch (e) {
+    console.log(`Failed to load edition [${edition}]`);
     // Catch all errors and simply assume an inactive edition which should not be viewable anywhere
-    return {active: false};
+    return {
+      edition,
+      active: false
+    };
```
Comment (jamesmorgan) [inline src/store/modules/kodaV2.js]: keep

File: src/utils.js [hunk 13]
```diff
@@ -48,173 +48,14 @@ const getEtherscanAddress = () => {
...
-
-  const HIGH_RES_EDITIONS = [
-    ...V1Editions,
-    ...V2Editions,
-  ];
+const isHighRes = (ipfsData, editionNumber) => {
+  const noHighRes = [19000];
+  if (noHighRes.indexOf(editionNumber) !== -1) {
+    return false;
+  }
```
Comment (jamesmorgan) [inline src/utils.js]: keep

File: src/utils.js [hunk 14]
```diff
@@ -48,173 +48,14 @@ const getEtherscanAddress = () => {
...
 
-  return _.includes(HIGH_RES_EDITIONS, editionNumber);
+  let tags = _.get(ipfsData.attributes, 'tags', []);
+  return _.some(tags, (element) => _.indexOf(['high res', 'High res'], element) >= 0);
```
Comment (jamesmorgan) [inline src/utils.js]: keep

**Pull request closed without merging**